pybamm/parameters/lithium_ion_parameters.py

```diff
@@ -791,24 +502,32 @@ def _set_dimensional_parameters(self):
                 ),
             },
         )
-        self.c_init = self.c_init_dimensional / self.c_max
         self.c_init_av = pybamm.xyz_average(pybamm.r_average(self.c_init))
+        self.sto_init_av = self.c_init_av / self.c_max
         eps_c_init_av = pybamm.xyz_average(
             self.epsilon_s * pybamm.r_average(self.c_init)
         )
-        self.n_Li_init = eps_c_init_av * self.c_max * self.domain_param.L * main.A_cc
+        self.n_Li_init = eps_c_init_av * self.domain_param.L * main.A_cc
         self.Q_Li_init = self.n_Li_init * main.F / 3600
 
-        eps_s_av = pybamm.xyz_average(self.epsilon_s)
-        self.elec_loading = eps_s_av * self.domain_param.L * self.c_max * main.F / 3600
+        self.epsilon_s_av = pybamm.xyz_average(self.epsilon_s)
+        self.elec_loading = (
+            self.epsilon_s_av * self.domain_param.L * self.c_max * main.F / 3600
+        )
         self.Q_init = self.elec_loading * main.A_cc
 
-        self.U_init_dim = self.U_dimensional(self.c_init_av, main.T_init_dim)
+        self.U_init = self.U(self.sto_init_av, main.T_init)
+
+        if main.options["particle shape"] == "spherical":
+            self.a_typ = 3 * pybamm.xyz_average(self.epsilon_s) / self.R_typ
 
-    def D_dimensional(self, sto, T):
+    def D(self, c_s, T):
         """Dimensional diffusivity in particle. Note this is defined as a
```

Comment:
Can the docstring clarify that the function in `ParameterValues` is defined as a function of `sto`, but here `D` is a function of `c_s`?